spec/controllers/course/discussion/topics_controller_spec.rb

```diff
@@ -17,80 +17,125 @@
+        end
+      end
+
+      context 'when a course student visits the page' do
+        before { sign_in(student) }
+
+        it { expect { subject }.to raise_exception(CanCan::AccessDenied) }
       end
     end
 
     describe '#my_students' do
       subject { get :my_students, course_id: course }
 
-      it { is_expected.to render_template(:my_students) }
+      context 'when a course staff visits the page' do
+        before { sign_in(staff) }
 
-      context 'when the user does not have any students' do
-        it 'shows nothing' do
-          subject
-          expect(topics).to be_empty
-        end
-      end
+        it { is_expected.to render_template(:my_students) }
 
-      context 'when the user has students' do
-        before do
-          group = create(:course_group, course: course, creator: user)
-          student_course_user = course.course_users.find_by(user_id: student)
-          create(:course_group_user, course: course, group: group, course_user: student_course_user)
+        context 'when the user does not have any students' do
+          it 'shows nothing' do
+            subject
+            expect(topics).to be_empty
+          end
         end
 
-        it "shows my students' topics" do
-          subject
-          expect(topics).to contain_exactly(topic, pending_topic)
+        context 'when the user has students' do
+          before do
+            group = create(:course_group, course: course, creator: staff)
+            student_course_user = course.course_users.find_by(user_id: student)
+            create(:course_group_user,
+                   course: course, group: group, course_user: student_course_user)
+          end
+
+          it "shows my students' topics" do
+            subject
+            expect(topics).to contain_exactly(topic, pending_topic)
+          end
         end
       end
+
+      context 'when a course student visits the page' do
+        before { sign_in(student) }
+
+        it { expect { subject }.to raise_exception(CanCan::AccessDenied) }
+      end
     end
 
     describe '#my_students_pending' do
       subject { get :my_students_pending, course_id: course }
 
-      it { is_expected.to render_template(:my_students_pending) }
+      context 'when a course staff visits the page' do
+        before { sign_in(staff) }
 
-      context 'when the user does not have any students' do
-        it 'shows nothing' do
-          subject
-          expect(topics).to be_empty
-        end
-      end
+        it { is_expected.to render_template(:my_students_pending) }
 
-      context 'when the user has students' do
-        before do
-          group = create(:course_group, course: course, creator: user)
-          student_course_user = course.course_users.find_by(user_id: student)
-          create(:course_group_user, course: course, group: group, course_user: student_course_user)
+        context 'when the user does not have any students' do
+          it 'shows nothing' do
```

Comment:
I not very keen to do that because for TA the default page from sidebar is my students page.